brokerapi/brokers/cloudsql/broker.go

```diff
@@ -429,7 +428,7 @@ func (b *CloudSQLBroker) ensureUsernamePassword(instanceID, bindingID string, de
 // Bind creates a new username, password, and set of ssl certs for the given instance.
 // The function may be slow to return because CloudSQL operations are asynchronous.
 // The default PCF service broker timeout may need to be raised to 90 or 120 seconds to accommodate the long bind time.
-func (b *CloudSQLBroker) Bind(instanceID, bindingID string, details brokerapi.BindDetails) (models.ServiceBindingCredentials, error) {
+func (b *CloudSQLBroker) Bind(ctx context.Context, instanceID, bindingID string, details brokerapi.BindDetails) (models.ServiceBindingCredentials, error) {
 
 	cloudDb, err := db_service.GetServiceInstanceDetailsById(instanceID)
```

Comment:
I think for consistency's sake it would also make sense to pass the ctx through to the db_service function calls. The fact that it uses GORM is an implementation detail which the rest of the system shouldn't be aware of. My experience with Go code bases is if you're going to use ctx, then it's best practice to pass it to all downstream calls.

Separately, it looks like GORM does not currently supports contexts, although they have an open feature request for it: https://github.com/jinzhu/gorm/issues/1231 which would make a case for pre-positioning this code base to take advantage of it when it does get implemented.

Reply:
Good idea, done!